helm/cluster/templates/clusterapi/controlplane/_helpers_prekubeadmcommands.tpl

```diff
@@ -9,10 +9,18 @@
 */}}
 {{- define "cluster.internal.controlPlane.kubeadm.preKubeadmCommands" }}
 {{- include "cluster.internal.kubeadm.preKubeadmCommands" $ }}
+{{- include "cluster.internal.controlPlane.kubeadm.preKubeadmCommands.default" $ }}
 {{- include "cluster.internal.controlPlane.kubeadm.preKubeadmCommands.provider" $ }}
 {{- include "cluster.internal.controlPlane.kubeadm.preKubeadmCommands.custom" $ }}
 {{- end }}
 
+{{/* Default commands to run before kubeadm on control plane nodes */}}
+{{- define "cluster.internal.controlPlane.kubeadm.preKubeadmCommands.default" }}
+- /opt/bin/setup-apiserver-environment.sh
+- env $(cat /etc/apiserver-environment | xargs) envsubst < /etc/kubeadm.yml > /etc/kubeadm.yml.tmp
+- mv /etc/kubeadm.yml.tmp /etc/kubeadm.yml
```

Comment:
do we need to change the file permission or overwriting owner / group again or is this ok?

Reply:
It's ok, the original file as well is owned by root (these are the same instructions we use in [the flatcarKubeadmPreCommands in cluster-aws](https://github.com/giantswarm/cluster-aws/blob/13e88385966bf2ac69b076a819b84f4a8975804a/helm/cluster-aws/templates/_helpers.tpl#L332)